### PR TITLE
chore: Add lint check for Artifactory paths

### PR DESCRIPTION
**Pull request by NicholasBoll**
Original PR: #11 

## Summary

* Added lint check for Artifactory paths
* Added linting to travis CI
* Added `frozen-lockfile` option to installs to prevent accidental yarn.lock file overriding
## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code has been documented and, if applicable, usage described in README.md
- [x] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [x] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: utils/check-lockfile-links.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const contents = fs.readFileSync(path.resolve(__dirname, '../yarn.lock')).toString();
+
+if (/resolved\s".*(artifactory).*\n/.test(contents)) {
+  console.error(
+    `This 'yarn.lock' file contains resolved urls that are not allowed. This usually happens if you're trying to install npm packages through a proxy like Artifactory. Make sure your npm registry is set to the publicly available one and try running 'yarn install' again.`
```
Comment (lychyi) [inline utils/check-lockfile-links.js]: Oops, this part sounds funny:

```npm registry is set to the publicly available one and try running 'yarn install' again.```

let's change it to

```npm registry is set to one that is publicly available and try running 'yarn install' again.```

Comment (NicholasBoll) [reply utils/check-lockfile-links.js]: Have you tried using the suggestion feature?

```suggestion
    `This 'yarn.lock' file contains resolved urls that are not allowed. This usually happens if you're trying to install npm packages through a proxy like Artifactory. Make sure your npm registry is set to one that is publicly available and try running 'yarn install' again. `
```

Comment (NicholasBoll) [reply utils/check-lockfile-links.js]: I guess that doesn't quite work with a rebase...

**Comment by lychyi** [comment]
I'm still trying to wrap my head around all the format/lint scripts we have in `package.json` and what they're for. We can tidy that up later though.

Last change that I forgot about. I would suggest adding this script in `lint-staged` so that this can't even be committed in the first place. Something like:

```
// package.json in "lint-staged" > "linters" 
"yarn.lock": [
  "node utils/check-lockfile-links.js",
  "git add"
],
```

**Comment by NicholasBoll** [comment]
> Last change that I forgot about. I would suggest adding this script in `lint-staged` so that this can't even be committed in the first place. Something like:
> 
> ```
> // package.json in "lint-staged" > "linters" 
> "yarn.lock": [
>   "node utils/check-lockfile-links.js",
>   "git add"
> ],
> ```

Good call. I'll add that.

**Pull request merged**